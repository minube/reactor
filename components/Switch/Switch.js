import { func, string } from 'prop-types';
import React from 'react';
import { Switch as SwitchNative, View } from 'react-native';

import { THEME } from '../../common';
import InputLabel from '../Input/InputLabel';
import styles from './Switch.style';

const { COLOR } = THEME;

const Switch = ({
  color, label, onChange, ...inherit
}) => (
  <View style={[styles.container, inherit.style]}>
    <SwitchNative
      {...inherit}
      value={inherit.value || false}
      activeThumbColor={color || COLOR.PRIMARY}
      activeTrackColor={COLOR.BASE}
      onValueChange={onChange}
      style={styles.switch}
    />
    { label && (
      <InputLabel>
        {label}
      </InputLabel>)}
  </View>
);

Switch.propTypes = {
  color: string,
  label: string,
  onChange: func,
};

Switch.defaultProps = {
  color: undefined,
  label: undefined,
  onChange: undefined,
};

export default Switch;
