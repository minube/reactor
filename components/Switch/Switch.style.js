import { StyleSheet } from 'react-native';

import { STYLE, THEME } from '../../common';

const { UNIT, OFFSET } = THEME;

export default StyleSheet.create({

  container: {
    ...STYLE.ROW,
    height: UNIT * 4.4,
    marginBottom: OFFSET,
  },

  label: {
    marginRight: UNIT,
  },

  switch: {
    height: UNIT * 2.4,
  },
});
