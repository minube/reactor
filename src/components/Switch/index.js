import { array, number, oneOfType, string } from 'prop-types';
import React from 'react';
import { Switch, Text, View } from 'react-native';

import { STYLE, THEME } from '../../common';
import styles from './style';

const { COLOR } = THEME;

const SwitchInput = ({ label, style, ...inherit }) => (
  <View style={[STYLE.ROW, styles.container, style]}>
    { label && <Text style={[styles.label, (inherit.disabled && styles.disabled)]}>{label}</Text>}
    <Switch
      {...inherit}
      activeThumbColor={COLOR.PRIMARY}
      activeTrackColor={COLOR.BORDER}
      style={styles.switch}
    />
  </View>
);

SwitchInput.propTypes = {
  label: string,
  style: oneOfType([array, number]),
};

SwitchInput.defaultProps = {
  label: undefined,
  style: [],
};

export default SwitchInput;
