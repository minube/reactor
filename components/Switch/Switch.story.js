import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text, boolean } from '@storybook/addon-knobs/react';

import WebView from '../WebView';
import Switch from './Switch';

storiesOf('✅ Switch', module)
  .addWithJSX('default', () => (
    <Switch />
  ))
  .addWithJSX('label', () => (
    <Switch label="Example of label" />
  ))
  .addWithJSX('disabled', () => (
    <Switch disabled label="Example of label" />
  ))
  .addWithJSX('style', () => (
    <Switch style={{ backgroundColor: 'orange' }} />
  ))
  .addWithJSX('🏀 Playground', () => (
    <WebView>
      <Switch
        disabled={boolean('disabled', false)}
        label={text('label', 'Example of label')}
        value={boolean('value', true)}
        onValueChange={action('Switch.onValueChange()')}
      />
    </WebView>
  ));

