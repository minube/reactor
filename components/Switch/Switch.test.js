import React from 'react';
import renderer from 'react-test-renderer';

import Switch from './Switch';

describe('<Switch>', () => {
  it('renders', () => {
    const tree = renderer.create(<Switch />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {color}', () => {
    const tree = renderer.create(<Switch color="green" value />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {label}', () => {
    const tree = renderer.create(<Switch label="Hello world" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {disabled}', () => {
    const tree = renderer.create(<Switch visible />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<Switch style={{ backgroundColor: 'orange' }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
