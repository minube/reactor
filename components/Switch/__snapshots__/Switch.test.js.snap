// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch> renders 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "flex-end",
        "display": "flex",
        "flexDirection": "row",
        "height": 44,
        "marginBottom": 22,
      },
      undefined,
    ]
  }
>
  <RCTSwitch
    activeThumbColor="#ffffff"
    activeTrackColor="#10c0bc"
    disabled={false}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onValueChange={undefined}
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "height": 28,
        },
      ]
    }
    thumbColor="#ffffff"
    trackColor="#cdd2d4"
    value={false}
  />
</View>
`;

exports[`<Switch> when {color} 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "flex-end",
        "display": "flex",
        "flexDirection": "row",
        "height": 44,
        "marginBottom": 22,
      },
      undefined,
    ]
  }
>
  <RCTSwitch
    activeThumbColor="#ffffff"
    activeTrackColor="green"
    disabled={false}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onValueChange={undefined}
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "height": 28,
        },
      ]
    }
    thumbColor="#ffffff"
    trackColor="#cdd2d4"
    value={true}
  />
</View>
`;

exports[`<Switch> when {disabled} 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "flex-end",
        "display": "flex",
        "flexDirection": "row",
        "height": 44,
        "marginBottom": 22,
      },
      undefined,
    ]
  }
>
  <RCTSwitch
    activeThumbColor="#ffffff"
    activeTrackColor="#10c0bc"
    disabled={false}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onValueChange={undefined}
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "height": 28,
        },
      ]
    }
    thumbColor="#ffffff"
    trackColor="#cdd2d4"
    value={false}
    visible={true}
  />
</View>
`;

exports[`<Switch> when {label} 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "flex-end",
        "display": "flex",
        "flexDirection": "row",
        "height": 44,
        "marginBottom": 22,
      },
      undefined,
    ]
  }
>
  <RCTSwitch
    activeThumbColor="#ffffff"
    activeTrackColor="#10c0bc"
    disabled={false}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onValueChange={undefined}
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "height": 28,
        },
      ]
    }
    thumbColor="#ffffff"
    trackColor="#cdd2d4"
    value={false}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontFamily": undefined,
          "fontSize": 14,
          "fontWeight": "600",
          "lineHeight": 16,
        },
        Object {
          "color": "#707070",
        },
        Object {
          "marginBottom": 0,
          "marginLeft": 10,
        },
      ]
    }
  >
    Hello world
  </Text>
</View>
`;

exports[`<Switch> when {style} 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "flex-end",
        "display": "flex",
        "flexDirection": "row",
        "height": 44,
        "marginBottom": 22,
      },
      Object {
        "backgroundColor": "orange",
      },
    ]
  }
>
  <RCTSwitch
    activeThumbColor="#ffffff"
    activeTrackColor="#10c0bc"
    disabled={false}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onValueChange={undefined}
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "height": 28,
        },
      ]
    }
    thumbColor="#ffffff"
    trackColor="#cdd2d4"
    value={false}
  />
</View>
`;
